### PR TITLE
feat: Wire up favicon form

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Wires up Team Settings > Design Setting > Favicon
 - Updates favicon in `/preview` and `/unpublished` routes if one is set in DB for team

**To test** 
- Standard "X" favicon still appears in Editor and for "unset" teams
- If I upload a favicon for a team and save, it displays in `/preview` and `/unpublished` routes 
- File type restrictions work

image

image

Demo: https://2680.planx.pizza/barking-and-dagenham/backup-dec23-article4-barkinganddagenham-manual-constraints/preview?analytics=false

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/FaviconForm.tsx [hunk 1]
```diff
@@ -39,8 +60,16 @@ export const FaviconForm: React.FC = () => {
       input={
         <InputRow>
           <InputRowLabel>Favicon:</InputRowLabel>
-          <InputRowItem width={50}>
-            <PublicFileUploadButton />
+          <InputRowItem width={formik.values.favicon ? 90 : 50}>
+            <ImgInput
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/FaviconForm.tsx:64]: I've updated this to use the wrapper `ImgInput` component which gives us a preview and a way to remove the image.

I'll open up a very small follow-on PR to make this change to the Logo form shortly to keep this visually and behaviorally consistent.

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/FaviconForm.tsx [hunk 2]
```diff
@@ -39,8 +60,16 @@ export const FaviconForm: React.FC = () => {
       input={
         <InputRow>
           <InputRowLabel>Favicon:</InputRowLabel>
-          <InputRowItem width={50}>
-            <PublicFileUploadButton />
+          <InputRowItem width={formik.values.favicon ? 90 : 50}>
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/FaviconForm.tsx:63]: Resizing this on upload looks better to me than keeping it wide the whole time - very open to opinions here if this jump in not consistent / a bit weird @ianjon3s

File: editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/index.tsx [hunk 3]
```diff
@@ -139,7 +139,7 @@ const DesignSettings: React.FC = () => {
           <ThemeAndLogoForm team={team} onSuccess={() => setOpen(true)} />
           <ButtonForm team={team} onSuccess={() => setOpen(true)} />
           <TextLinkForm team={team} onSuccess={() => setOpen(true)} />
-          <FaviconForm />
+          <FaviconForm team={team} onSuccess={() => setOpen(true)} />
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Settings/DesignSettings/index.tsx:142]: There's a little repetition with some of these props and form behaviours - I'll look at a small follow on PR to try and tidy this up 👌

**Review by Mike-Heneghan — APPROVED** [review]
Working as expected for me, looks great 👍

**Pull request merged**